Fedora testing and warnings

TheNodi opened the pull request:
I've tested the package on Fedora 23-25 and everything it's working as expected.

I've wrapped the `Nginx::stop();` inside installation in a try-catch because it was failing if nginx wasn't available (it couldn't find the service).

I've added a Warning class to display common problems upon installation, so it's easier to spot a possible misconfiguration of the system. Should I add a "Continue? [Y/n]" prompt in case of warnings?

cpriego commented:
I think we should just stop in the case of those particular warnings (maybe after checking for both). If SELinux is enabled or we are in /root it'll break the installation anyway.

TheNodi commented:
I've changed warnings to be runtime exceptions and stop the installation.

Since SELinux can be configured to allow that kind of setup, I've added a `--ignore-selinux` flag to skip that check.

Tests are included, half of them at least since the use of a constant is kind difficult to test. There's a `runkit_constant_remove` function but I don't really like that approach. Maybe valet's home can be moved to a container variable.
